pkg/localdata/profile.go

```diff
@@ -321,6 +328,60 @@ func (p *Profile) SelectInstalledVersion(component string, version v0manifest.Ve
 	return version, nil
 }
 
+// ResetMirror reset root.json and cleanup manifests directory
+func (p *Profile) ResetMirror(addr, root string, forceRefresh bool) error {
+	shaWriter := sha256.New()
+	if _, err := io.Copy(shaWriter, strings.NewReader(addr)); err != nil {
+		return err
+	}
+	localRoot := p.Path("bin", fmt.Sprintf("%s.root.json", hex.EncodeToString(shaWriter.Sum(nil))[:16]))
+
+	if root == "" {
+		if utils.IsExist(localRoot) && !forceRefresh {
+			root = localRoot
+		} else if strings.HasSuffix(addr, "/") {
+			root = addr + "root.json"
+		} else {
+			root = addr + "/root.json"
+		}
+	}
+
+	var wc io.ReadCloser
+	if strings.HasPrefix(root, "http") {
+		if resp, err := http.Get(root); err == nil {
```

Comment:
Return an error if the mirror starts with `http` and `root == "" && forceRefresh == false`.